plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/GradleUpdateChecker.kt

```diff
@@ -0,0 +1,44 @@
+package de.fayard.refreshVersions.core.internal
+
+import com.squareup.moshi.Moshi
+import de.fayard.refreshVersions.core.extensions.okhttp.await
+import okhttp3.OkHttpClient
+import okhttp3.Request
+
+internal class GradleUpdateChecker(
+    val httpClient: OkHttpClient,
+    val moshi: Moshi
+) {
+    val API_URL = "https://services.gradle.org/versions/current"
+
+    suspend fun fetchGradleCurrentVersion(): GradleCurrentVersion? {
+        val request = Request.Builder().url(API_URL).build()
+        return httpClient.newCall(request).await().use { response ->
+            if (response.isSuccessful) {
+                moshi.adapter(GradleCurrentVersion::class.java)
+                    .fromJson(response.body!!.source())
+            } else {
+                null
```

Comment:
Why would we let `IOException`s and json data exceptions throw while we ignore silently non 2xx responses from the server such as 503?

For the time being, I'd throw an `HttpException` (from retrofit2) for consistency, until we improve error reporting and tolerance (which has implications in parsing and writing the `versions.properties` file.